test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1

```diff
@@ -822,6 +822,24 @@ Describe "Invoke-WebRequest tests" -Tags "Feature" {
         $response.Content.Method | Should -Be $redirectedMethod
     }
 
+    It "Validates Invoke-WebRequest handles responses without Location header correctly for redirect: <redirectType>" -TestCases $redirectTests {
+        param($redirectType, $redirectedMethod)
+        # Skip relative test as it is not a valid response type.
+        if ($redirectType -eq 'relative'){ return }
+
+        # PowerShell should throw an HTTP Response Exception
+        # for a redirect response which does not contain a Location header
+        # the correct redirect status code should be included in the exception message
```

Comment:
We should include a note about  this requiring a Authorization header. Something like

```powershell
# When an Authorization request header is present,
# and -PreserveAuthorizationOnRedirect is not present,
# PowerShell should throw an HTTP Response Exception
# for a redirect response which does not contain a Location response header.
# The correct redirect status code should be included in the exception.
```

It's a mouth full, but at least people will understand why we are doing what seems like a superfluous set of tests.